02-peer-protocol.md

```diff
@@ -2773,18 +3387,33 @@ A node:
 
 A receiving node:
   - if `next_funding_txid` is set:
-    - if `next_funding_txid` matches the latest interactive funding transaction:
-      - if it has not received `tx_signatures` for that funding transaction:
+    - if `next_funding_txid` matches the latest interactive funding transaction
+      or the current channel funding transaction:
+      - if `next_commitment_number` is equal to the commitment number of the
+        `commitment_signed` message it sent for this funding transaction:
         - MUST retransmit its `commitment_signed` for that funding transaction.
-        - if it has already received `commitment_signed` and it should sign first,
-          as specified in the [`tx_signatures` requirements](#the-tx_signatures-message):
-          - MUST send its `tx_signatures` for that funding transaction.
+      - if it has already received `commitment_signed` and it should sign first,
+        as specified in the [`tx_signatures` requirements](#the-tx_signatures-message):
+        - MUST send its `tx_signatures` for that funding transaction.
       - if it has already received `tx_signatures` for that funding transaction:
         - MUST send its `tx_signatures` for that funding transaction.
+    - if it also sets `next_funding_txid` in its own `channel_reestablish`, but the
+      values don't match:
+      - MUST send an `error` and fail the channel.
     - otherwise:
       - MUST send `tx_abort` to let the sending node know that they can forget
         this funding transaction.
 
+A receiving node:
+  - if `my_current_funding_locked` does not match the most recent `splice_locked`
+    it has received:
+    - MUST process `my_current_funding_locked` as if it was receiving `splice_locked`
+      for this `txid`, and thus discard the previous funding transaction and RBF
+      attempts if it has previously sent its own `splice_locked` for that `txid`.
+  - if `your_last_funding_locked` is not set, or if it does not match the most recent
```

Comment:
`your_last_funding_locked` should be the original funding txid in this case and not NULL.

Reply:
Ah yeah that should be fine for `channey_ready`. Sending `splice_locked` if `your_last_funding_locked` is `NULL` is the issue -- I don't think there should be any case where that should happen.

Maybe that is something that comes up in 0-conf?

I would imagine the spec should read:
```
  - if `your_last_funding_locked` is set and it does not match the most recent
    `splice_locked` it has sent:
    - MUST retransmit `splice_locked`.
```

Leaving the NULL `your_last_funding_locked` cases to be handled by `channel_ready` seperately